Analytical_Engine/Query/IsVertical.cs

```diff
@@ -0,0 +1,61 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model(BHoM)
+ * Copyright(c) 2015 - 2020, the respective contributors.All rights reserved.
+
+*
+* Each contributor holds copyright over their respective contributions.
+
+* The project versioning(Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or
+* (at your option) any later version.
+
+*
+* The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.See the
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this code.If not, see<https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.oM.Analytical.Elements;
+using BH.oM.Geometry;
+using BH.oM.Reflection.Attributes;
+using BH.Engine.Geometry;
+using BH.Engine.Reflection;
+
+
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.ComponentModel;
+
+namespace BH.Engine.Analytical
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Determines whether a panel is vertical.")]
+        [Input("panel", "The IPanel to check if a panel is vertical.")]
+        [Output("bool", "True for panels for panels is vertical.")]
+        public static bool IsVertical<TEdge, TOpening>(this IPanel<TEdge, TOpening> panel)
+            where TEdge : IEdge
+            where TOpening : IOpening<TEdge>
+        {
+            if (IsAligned(panel, Plane.XZ) == false)
+                return IsAligned(panel, Plane.YZ);
+            else
+                return IsAligned(panel, Plane.XZ);
```

Comment:
I don't think we can just check whether it is aligned with global XY and XZ, you could have a curve that exists in a plane at 45o to either of those, it would still be vertical but the method would return false.

I think what we need to do is check the curve is planar, get the tangent and make sure it is aligned to the vertical i.e. `Vector.ZAxis`.